integrations/gitlab_v2/main.py

```diff
@@ -0,0 +1,157 @@
+    return GitlabClient(
+        ocean.integration_config["gitlab_host"],
+        gitlab_access_token,
+    )
+
+
+async def bootstrap_client() -> None:
+    app_host = ocean.integration_config["app_host"]
+    tokens = ocean.integration_config["gitlab_access_tokens"]["tokens"]
+    if not app_host:
+        logger.warning(
+            "No app host provided, skipping webhook creation. "
+            "Without setting up the webhook, the integration will not export live changes from Gitlab"
+        )
+        return
+
+    if ocean.event_listener_type == "ONCE":
+        logger.info("Skipping webhook creation because the event listener is ONCE")
+        return
+
+    gitlab_client = initialize_client(tokens[0])
+    await gitlab_client.create_webhooks(app_host)
+
+
+async def handle_webhook_event(
+        webhook_event: str,
+        object_attributes_action: str,
+        data: dict[str, Any],
+) -> dict[str, Any]:
+    ocean_action = None
+    if object_attributes_action in DELETE_WEBHOOK_EVENTS:
+        ocean_action = ocean.unregister_raw
+    elif object_attributes_action in CREATE_UPDATE_WEBHOOK_EVENTS:
+        ocean_action = ocean.register_raw
+
+    if not ocean_action:
+        logger.info(f"Webhook event '{webhook_event}' not recognized.")
+        return {"ok": True}
+
+    if webhook_event == WebHookEventType.MERGE_REQUEST:
+        payload = extract_merge_request_payload(data)
+        await ocean_action(ResourceKind.MERGE_REQUEST, [payload])
+    elif webhook_event == WebHookEventType.ISSUE:
+        payload = extract_issue_payload(data)
+        logger.info(f"Upserting issue with payload: {payload}")
+        await ocean_action(ResourceKind.ISSUE, [payload])
+    else:
+        logger.info(f"Unhandled webhook event type: {webhook_event}")
+        return {"ok": True}
+
+    logger.info(f"Webhook event '{webhook_event}' processed successfully.")
+    return {"ok": True}
+
+
+@ocean.router.post("/webhook")
+async def handle_webhook_request(data: dict[str, Any]) -> dict[str, Any]:
+    webhook_event = data.get("event_type", "")
+    object_attributes_action = data.get("object_attributes", {}).get("action", "")
+    logger.info(
+        f"Received webhook event: {webhook_event} with action: {object_attributes_action}"
+    )
+
+    return await handle_webhook_event(webhook_event, object_attributes_action, data)
+
+
+@ocean.on_resync(ResourceKind.PROJECT)
+async def resync_project(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    tokens = ocean.integration_config["gitlab_access_tokens"]["tokens"]
+    for token in tokens:
+        client = initialize_client(token)
+        config = typing.cast(GitlabProjectResourceConfig, event.resource_config)
+
+        async for projects in client.get_projects():
+            logger.info(f"Received {kind} batch with {len(projects)} projects")
+            if config.selector.onlyGrouped:
+                projects = [project for project in projects if project.get("__group")]
```

Comment:
I don't see what this filter is solving?

Reply:
Issue has been addressed and this filter has been removed as my implementation now works with a single resync method as suggested by other reviewer.